kivy/core/image/__init__.py

```diff
@@ -862,12 +878,24 @@ def save(self, filename, flipped=False):
 
         l_pixels = len(pixels)
         if l_pixels == size[0] * size[1] * 3:
-            fmt = 'rgb'
+            pixelfmt = 'rgb'
         elif l_pixels == size[0] * size[1] * 4:
-            fmt = 'rgba'
+            pixelfmt = 'rgba'
         else:
             raise Exception('Unable to determine the format of the pixels')
-        return loader.save(filename, size[0], size[1], fmt, pixels, flipped)
+        return loader.save(filename, size[0], size[1], pixelfmt, pixels, flipped, fmt)
+
+    def _find_format_from_filename(self, filename):
+        ext = filename.rsplit(".", 1)[-1].lower()
+        if ext in (
+            'bmp', 'jpe', 'lbm', 'pcx', 'png', 'pnm',
+            'tga', 'tiff', 'webp', 'xcf', 'xpm', 'xv'):
```

Comment:
Minor taste suggestion: I find it more sexy to use sets for that type of things:
```python
        if ext in {
            'bmp', 'jpe', 'lbm', 'pcx', 'png', 'pnm',
            'tga', 'tiff', 'webp', 'xcf', 'xpm', 'xv'}:
```
Specially since it's `O(1)`